bundle/edu.gemini.itc/src/main/java/edu/gemini/itc/base/SEDFactory.java

```diff
@@ -143,10 +143,22 @@ private static VisitableSampledSpectrum getSED(final SourceDefinition sdp, final
         } else if (sdp.distribution() instanceof UserDefinedSpectrum) {
             final UserDefinedSpectrum userDefined = (UserDefinedSpectrum) sdp.distribution();
             final MagnitudeBand band = sdp.normBand();
-            // The user-supplied SED must cover the range of the instrument configuration and the normalization band.
+            Double inst_start_wave = instrument.getObservingStart();
+            Double inst_end_wave = instrument.getObservingEnd();
+
+            // GNIRS overrides getObservingStart and getObservingEnd for each order,
+            // so use the full GNIRS XD wavelength range:
+            if (instrument instanceof Gnirs) {
+                if (((Gnirs) instrument).XDisp_IsUsed()) {
+                    inst_start_wave = 750.;
+                    inst_end_wave = 2600.;
```

Comment:
N.B. the `123.` syntax for doubles is legal Java, but it's weird. `123.0` is more idiomatic.